qos/evm/store.go

```diff
@@ -49,56 +49,57 @@ type EndpointStore struct {
 	blockHeight uint64
 }
 
-// TODO_UPNEXT(@adshmh): Update this method along with the relayer.EndpointSelector interface.
-func (es *EndpointStore) Select(availableEndpoints map[relayer.AppAddr][]relayer.Endpoint) (relayer.AppAddr, relayer.EndpointAddr, error) {
+// Select returns an endpoint address matching an entry from the list of available endpoints.
+// available endpoints are filtered based on their validity first.
+// A random endpoint is then returned from the filtered list of valid endpoints.
+func (es *EndpointStore) Select(availableEndpoints []relayer.Endpoint) (relayer.EndpointAddr, error) {
+	filteredEndpointsAddr, err := es.filterEndpoints(availableEndpoints)
+	if err != nil {
+		return relayer.EndpointAddr(""), err
+	}
+
+	logger := es.Logger.With("number of available endpoints", len(availableEndpoints))
```

Comment:
It's a bit weird to log this way with the logger state. I would pseudo like this:

// Log number of available endpoints. About to filter them
// Filter
// Filtered n of m endpoints for use
// If n == 0 => log no good endpoints
// Select random endpoint